Improve logcollector configuration options

## Description

In case it is not required to specify a type of event to be collected, logcollector should take as many events as possible.

Current configuration. (for Windows)

```yml
logcollector:
  enabled: true
  reload_interval: 1m
  windows:
    - channel: System
      query: Event[System/EventID = 37]
```
The idea is that the query field is optional, and that a configuration like the following is valid.

```yml
logcollector:
  enabled: true
  reload_interval: 1m
  windows:
    - channel: System
```

## Definition of Done

- The `query` field must be optional.
- In case it does not find the query field, the collector must try to collect all types of events.
- Apply this logic to all available OS.
- Update documentation.

## Windows check 🟢 :

In each case the events generated where Application with ID 7654, System with 1234, and Security with 4625 with:

```console
PS C:\Windows\system32> Write-EventLog -LogName Application -Source "Application" -EventId 7654 -EntryType Information -Message "This is a test Application event"
PS C:\Windows\system32> Write-EventLog -LogName System -Source "EventLog" -EventId 1234 -EntryType Information -Message "This is a test System event"
PS C:\Windows\system32> Start-Process powershell -Credential $credential -ErrorAction SilentlyContinue
Start-Process : This command cannot be run due to the error: The user name or password is incorrect.
At line:1 char:1
+ Start-Process powershell -Credential $credential -ErrorAction Silentl ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Start-Process], InvalidOperationException
    + FullyQualifiedErrorId : InvalidOperationException,Microsoft.PowerShell.Commands.StartProcessCommand

PS C:\Windows\system32> Write-EventLog -LogName System -Source "EventLog" -EventId 1111 -EntryType Information -Message "This is a test System event"
PS C:\Windows\system32> Write-EventLog -LogName Application -Source "Application" -EventId 7777 -EntryType Information -Message "This is a test Application event"
```

* without the default query value:

config:
```yaml
  windows:
    - channel: System
      query: Event[System/EventID = 1234]
    - channel: Application
      query: Event[System/EventID = 7654]
    - channel: Security
      query: Event[System/EventID = 4625]
```

```console

C:\Program Files\wazuh-agent>wazuh-agent.exe --config-file C:\ProgramData\wazuh-agent\config\wazuh-agent.yml
[2025-04-03 11:20:17.998] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\agent\src\instance_handler_win.cpp:47] [getInstanceLock] Mutex created
[2025-04-03 11:20:18.000] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\agent\src\agent_runner.cpp:196] [StartAgent] Starting wazuh-agent
[2025-04-03 11:20:18.000] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\src\configuration_parser.cpp:46] [LoadLocalConfig] Loading local config file: C:\ProgramData\wazuh-agent\config\wazuh-agent.yml.
[2025-04-03 11:20:18.003] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:214] [GetConfig] Requested setting not found, default value used. Key not found: path.data
[2025-04-03 11:20:18.017] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:214] [GetConfig] Requested setting not found, default value used. Key not found: path.data
[2025-04-03 11:20:18.022] [wazuh-agent] [warning] [WARN] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:235] [GetParsedConfigInRangeOrDefault] Invalid range: min value is greater or equal to max value.
[2025-04-03 11:20:18.023] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:214] [GetConfig] Requested setting not found, default value used. Key not found: path.data
[2025-04-03 11:20:18.026] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\src\configuration_parser.cpp:97] [LoadSharedConfig] Loading shared configuration.
[2025-04-03 11:20:18.127] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:239] [PerformHttpRequest] Request /api/v1/authentication: Status 200
[2025-04-03 11:20:18.128] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:240] [PerformHttpRequest] Request endpoint: /api/v1/authentication
Response: HTTP/1.1 200 OK
X-Imposter-Request: 61345e98-5d1f-4d7d-ba66-b65c06ec1a05
Server: imposter
content-length: 316

{"token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJXYXp1aCIsImF1ZCI6IldhenVoIENvbW11bmljYXRpb25zIEFQSSIsImlhdCI6MTc0MzcwNDQxOSwiZXhwIjoxNzQzNzA1MzE5LCJ1dWlkIjoiNjE2MTllYTAtNTRhNi00MmUxLWFlNWQtY2NlNjQ3NWZhYTZiIn0.MEUCIGn6STzkyAjSFNJElORNzx_sHpiIZVGcQLtcsh9q6_TtAiEA4DsL5uPHp0tYNUGBUDXoToDnG6J3pxOWlDAvlwKCCFU"}
[2025-04-03 11:20:18.130] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\agent\communicator\src\communicator.cpp:109] [SendAuthenticationRequest] Successfully authenticated with the manager.
[2025-04-03 11:20:18.131] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:214] [GetConfig] Requested setting not found, default value used. Key not found: path.data
[2025-04-03 11:20:18.133] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:214] [GetConfig] Requested setting not found, default value used. Key not found: path.data
[2025-04-03 11:20:18.135] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:214] [GetConfig] Requested setting not found, default value used. Key not found: localfiles
[2025-04-03 11:20:18.136] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\inventory\src\inventory.cpp:14] [Start] Inventory module is disabled.
[2025-04-03 11:20:18.136] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:32] [Start] Logcollector module started.
[2025-04-03 11:20:18.137] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\file_reader\src\file_reader_win.cpp:27] [Reload] No matches found for pattern: /var/log/auth.log
[2025-04-03 11:20:18.138] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\winevt_reader\src\event_reader_win.cpp:79] [QueryEvents] Subscribed to events for 'System' channel with query: 'Event[System/EventID = 1234]'
[2025-04-03 11:20:18.139] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\winevt_reader\src\event_reader_win.cpp:79] [QueryEvents] Subscribed to events for 'Application' channel with query: 'Event[System/EventID = 7654]'
[2025-04-03 11:20:18.141] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\winevt_reader\src\event_reader_win.cpp:79] [QueryEvents] Subscribed to events for 'Security' channel with query: 'Event[System/EventID = 4625]'
[2025-04-03 11:20:18.217] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:162] [Co_PerformHttpRequest] Request /api/v1/commands: Status 200
[2025-04-03 11:20:18.218] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:163] [Co_PerformHttpRequest] Request endpoint: /api/v1/commands
Response: HTTP/1.1 200 OK
X-Imposter-Request: bb630cdd-4700-4ffd-90d8-033ba78bcd41
Server: imposter
content-length: 0


[2025-04-03 11:20:18.219] [wazuh-agent] [error] [ERROR] [C:\Users\wazuh-user\wazuh-agent\src\agent\task_manager\src\task_manager.cpp:110] [operator ()] FetchCommands coroutine task exited with an exception: [json.exception.parse_error.101] parse error at line 1, column 1: attempting to parse an empty input; check that your input string or stream contains the expected JSON
[2025-04-03 11:20:28.643] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:20:28.643] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:20:39.129] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:20:39.129] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:20:49.205] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:20:49.206] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:20:50.634] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:156] [SendMessageA] Message pushed: 'Application':'<Event xmlns='http://schemas.microsoft.com/win/2004/08/events/event'><System><Provider Name='Application'/><EventID Qualifiers='0'>7654</EventID><Version>0</Version><Level>4</Level><Task>1</Task><Opcode>0</Opcode><Keywords>0x80000000000000</Keywords><TimeCreated SystemTime='2025-04-03T18:20:50.6263244Z'/><EventRecordID>5322</EventRecordID><Correlation/><Execution ProcessID='824' ThreadID='0'/><Channel>Application</Channel><Computer>DESKTOP-D42041D</Computer><Security/></System><EventData><Data>This is a test Application event</Data></EventData></Event>'
[2025-04-03 11:20:59.237] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:20:59.237] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:21:10.561] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:156] [SendMessageA] Message pushed: 'System':'<Event xmlns='http://schemas.microsoft.com/win/2004/08/events/event'><System><Provider Name='EventLog'/><EventID Qualifiers='0'>1234</EventID><Version>0</Version><Level>4</Level><Task>1</Task><Opcode>0</Opcode><Keywords>0x80000000000000</Keywords><TimeCreated SystemTime='2025-04-03T18:21:10.5599584Z'/><EventRecordID>4892</EventRecordID><Correlation/><Execution ProcessID='824' ThreadID='0'/><Channel>System</Channel><Computer>DESKTOP-D42041D</Computer><Security/></System><EventData><Data>This is a test System event</Data></EventData></Event>'
[2025-04-03 11:21:16.730] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\communicator\src\communicator.cpp:384] [ExecuteRequestLoop] Items count: 2
[2025-04-03 11:21:17.230] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:162] [Co_PerformHttpRequest] Request /api/v1/events/stateless: Status 200
[2025-04-03 11:21:17.231] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:163] [Co_PerformHttpRequest] Request endpoint: /api/v1/events/stateless
Response: HTTP/1.1 200 OK
X-Imposter-Request: 74882ad8-9965-4c6f-a43d-6c43007a8508
Server: imposter
content-length: 0


[2025-04-03 11:21:18.988] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\file_reader\src\file_reader_win.cpp:27] [Reload] No matches found for pattern: /var/log/auth.log
[2025-04-03 11:21:26.799] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:21:28.323] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:21:32.355] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:156] [SendMessageA] Message pushed: 'Security':'<Event xmlns='http://schemas.microsoft.com/win/2004/08/events/event'><System><Provider Name='Microsoft-Windows-Security-Auditing' Guid='{54849625-5478-4994-a5ba-3e3b0328c30d}'/><EventID>4625</EventID><Version>0</Version><Level>0</Level><Task>12544</Task><Opcode>0</Opcode><Keywords>0x8010000000000000</Keywords><TimeCreated SystemTime='2025-04-03T18:21:31.3417682Z'/><EventRecordID>32238</EventRecordID><Correlation ActivityID='{da50e5bc-a4fc-0000-a7e6-50dafca4db01}'/><Execution ProcessID='732' ThreadID='6160'/><Channel>Security</Channel><Computer>DESKTOP-D42041D</Computer><Security/></System><EventData><Data Name='SubjectUserSid'>S-1-5-21-3444752446-47338941-2635532298-1001</Data><Data Name='SubjectUserName'>vagrant</Data><Data Name='SubjectDomainName'>DESKTOP-D42041D</Data><Data Name='SubjectLogonId'>0xa8d30</Data><Data Name='TargetUserSid'>S-1-0-0</Data><Data Name='TargetUserName'>vagrant</Data><Data Name='TargetDomainName'>DESKTOP-D42041D</Data><Data Name='Status'>0xc000006d</Data><Data Name='FailureReason'>%%2313</Data><Data Name='SubStatus'>0xc000006a</Data><Data Name='LogonType'>2</Data><Data Name='LogonProcessName'>seclogo</Data><Data Name='AuthenticationPackageName'>Negotiate</Data><Data Name='WorkstationName'>DESKTOP-D42041D</Data><Data Name='TransmittedServices'>-</Data><Data Name='LmPackageName'>-</Data><Data Name='KeyLength'>0</Data><Data Name='ProcessId'>0xb40</Data><Data Name='ProcessName'>C:\Windows\System32\svchost.exe</Data><Data Name='IpAddress'>::1</Data><Data Name='IpPort'>0</Data></EventData></Event>'
[2025-04-03 11:21:36.868] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:21:38.397] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:21:38.398] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\communicator\src\communicator.cpp:384] [ExecuteRequestLoop] Items count: 1
[2025-04-03 11:21:38.887] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:162] [Co_PerformHttpRequest] Request /api/v1/events/stateless: Status 200
[2025-04-03 11:21:38.888] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:163] [Co_PerformHttpRequest] Request endpoint: /api/v1/events/stateless
Response: HTTP/1.1 200 OK
X-Imposter-Request: 7adc20e3-661e-498b-aa8c-7b1e21fa970c
Server: imposter
content-length: 0


[2025-04-03 11:21:46.941] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:21:50.017] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:21:57.010] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:22:00.054] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:22:07.405] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:22:10.775] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:22:18.096] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:22:19.943] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\file_reader\src\file_reader_win.cpp:27] [Reload] No matches found for pattern: /var/log/auth.log
[2025-04-03 11:22:20.220] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\inventory\src\inventory.cpp:78] [Stop] Inventory module stopping...
[2025-04-03 11:22:20.220] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:204] [Wait] Logcollector coroutine timer was canceled.
[2025-04-03 11:22:20.222] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:204] [Wait] Logcollector coroutine timer was canceled.
[2025-04-03 11:22:20.223] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\winevt_reader\src\event_reader_win.cpp:86] [QueryEvents] Unsubscribing to channel 'System'.
[2025-04-03 11:22:20.223] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:204] [Wait] Logcollector coroutine timer was canceled.
[2025-04-03 11:22:20.223] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\winevt_reader\src\event_reader_win.cpp:86] [QueryEvents] Unsubscribing to channel 'Application'.
[2025-04-03 11:22:20.224] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:204] [Wait] Logcollector coroutine timer was canceled.
[2025-04-03 11:22:20.225] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\winevt_reader\src\event_reader_win.cpp:86] [QueryEvents] Unsubscribing to channel 'Security'.
[2025-04-03 11:22:20.235] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:101] [Stop] Logcollector module stopped.
```


* with the default query value:

config:
```yaml
  windows:
    - channel: System
    - channel: Application
    - channel: Security
```

```console
C:\Program Files\wazuh-agent>wazuh-agent.exe --config-file C:\ProgramData\wazuh-agent\config\wazuh-agent.yml
[2025-04-03 11:32:40.650] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\agent\src\instance_handler_win.cpp:47] [getInstanceLock] Mutex created
[2025-04-03 11:32:40.651] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\agent\src\agent_runner.cpp:196] [StartAgent] Starting wazuh-agent
[2025-04-03 11:32:40.652] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\src\configuration_parser.cpp:46] [LoadLocalConfig] Loading local config file: C:\ProgramData\wazuh-agent\config\wazuh-agent.yml.
[2025-04-03 11:32:40.653] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:214] [GetConfig] Requested setting not found, default value used. Key not found: path.data
[2025-04-03 11:32:40.664] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:214] [GetConfig] Requested setting not found, default value used. Key not found: path.data
[2025-04-03 11:32:40.667] [wazuh-agent] [warning] [WARN] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:235] [GetParsedConfigInRangeOrDefault] Invalid range: min value is greater or equal to max value.
[2025-04-03 11:32:40.667] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:214] [GetConfig] Requested setting not found, default value used. Key not found: path.data
[2025-04-03 11:32:40.670] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\src\configuration_parser.cpp:97] [LoadSharedConfig] Loading shared configuration.
[2025-04-03 11:32:41.162] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:239] [PerformHttpRequest] Request /api/v1/authentication: Status 200
[2025-04-03 11:32:41.163] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:240] [PerformHttpRequest] Request endpoint: /api/v1/authentication
Response: HTTP/1.1 200 OK
X-Imposter-Request: 1ef0a658-ebad-4e6a-8f62-8c756be72db5
Server: imposter
content-length: 315

{"token":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJXYXp1aCIsImF1ZCI6IldhenVoIENvbW11bmljYXRpb25zIEFQSSIsImlhdCI6MTc0MzcwNTE2MCwiZXhwIjoxNzQzNzA2MDYwLCJ1dWlkIjoiNjE2MTllYTAtNTRhNi00MmUxLWFlNWQtY2NlNjQ3NWZhYTZiIn0.MEQCIB3gZP9rS8WmcwMu1Bs35DJG9ukU5l4-pxTP7EFAWXnIAiBggEEnWjCzGZig3wbUhYIHT13j0-d2rFA05Z62eNqnoA"}
[2025-04-03 11:32:41.164] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\agent\communicator\src\communicator.cpp:109] [SendAuthenticationRequest] Successfully authenticated with the manager.
[2025-04-03 11:32:41.165] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:214] [GetConfig] Requested setting not found, default value used. Key not found: path.data
[2025-04-03 11:32:41.166] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:214] [GetConfig] Requested setting not found, default value used. Key not found: path.data
[2025-04-03 11:32:41.169] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\configuration_parser\include\configuration_parser.hpp:214] [GetConfig] Requested setting not found, default value used. Key not found: localfiles
[2025-04-03 11:32:41.170] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\inventory\src\inventory.cpp:14] [Start] Inventory module is disabled.
[2025-04-03 11:32:41.170] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:32] [Start] Logcollector module started.
[2025-04-03 11:32:41.171] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\file_reader\src\file_reader_win.cpp:27] [Reload] No matches found for pattern: /var/log/auth.log
[2025-04-03 11:32:41.173] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\winevt_reader\src\event_reader_win.cpp:79] [QueryEvents] Subscribed to events for 'System' channel with query: '*'
[2025-04-03 11:32:41.174] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\winevt_reader\src\event_reader_win.cpp:79] [QueryEvents] Subscribed to events for 'Application' channel with query: '*'
[2025-04-03 11:32:41.174] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\winevt_reader\src\event_reader_win.cpp:79] [QueryEvents] Subscribed to events for 'Security' channel with query: '*'
[2025-04-03 11:32:41.290] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:162] [Co_PerformHttpRequest] Request /api/v1/commands: Status 200
[2025-04-03 11:32:41.291] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:163] [Co_PerformHttpRequest] Request endpoint: /api/v1/commands
Response: HTTP/1.1 200 OK
X-Imposter-Request: 796fc026-0bb6-4691-860f-76a4ee6a17a1
Server: imposter
content-length: 0


[2025-04-03 11:32:41.292] [wazuh-agent] [error] [ERROR] [C:\Users\wazuh-user\wazuh-agent\src\agent\task_manager\src\task_manager.cpp:110] [operator ()] FetchCommands coroutine task exited with an exception: [json.exception.parse_error.101] parse error at line 1, column 1: attempting to parse an empty input; check that your input string or stream contains the expected JSON
[2025-04-03 11:32:51.172] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:32:51.173] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:32:58.661] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:156] [SendMessageA] Message pushed: 'System':'<Event xmlns='http://schemas.microsoft.com/win/2004/08/events/event'><System><Provider Name='EventLog'/><EventID Qualifiers='0'>1234</EventID><Version>0</Version><Level>4</Level><Task>1</Task><Opcode>0</Opcode><Keywords>0x80000000000000</Keywords><TimeCreated SystemTime='2025-04-03T18:32:58.6485494Z'/><EventRecordID>4898</EventRecordID><Correlation/><Execution ProcessID='824' ThreadID='0'/><Channel>System</Channel><Computer>DESKTOP-D42041D</Computer><Security/></System><EventData><Data>This is a test System event</Data></EventData></Event>'
[2025-04-03 11:33:01.234] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:33:01.234] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:33:01.234] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\communicator\src\communicator.cpp:384] [ExecuteRequestLoop] Items count: 1
[2025-04-03 11:33:01.547] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:162] [Co_PerformHttpRequest] Request /api/v1/events/stateless: Status 200
[2025-04-03 11:33:01.548] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:163] [Co_PerformHttpRequest] Request endpoint: /api/v1/events/stateless
Response: HTTP/1.1 200 OK
X-Imposter-Request: 41dc9138-28e6-4e81-88a5-a71569decb62
Server: imposter
content-length: 0


[2025-04-03 11:33:07.848] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:156] [SendMessageA] Message pushed: 'Application':'<Event xmlns='http://schemas.microsoft.com/win/2004/08/events/event'><System><Provider Name='Application'/><EventID Qualifiers='0'>7654</EventID><Version>0</Version><Level>4</Level><Task>1</Task><Opcode>0</Opcode><Keywords>0x80000000000000</Keywords><TimeCreated SystemTime='2025-04-03T18:33:07.8076064Z'/><EventRecordID>5324</EventRecordID><Correlation/><Execution ProcessID='824' ThreadID='0'/><Channel>Application</Channel><Computer>DESKTOP-D42041D</Computer><Security/></System><EventData><Data>This is a test Application event</Data></EventData></Event>'
[2025-04-03 11:33:11.343] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:33:12.655] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:33:12.656] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\communicator\src\communicator.cpp:384] [ExecuteRequestLoop] Items count: 1
[2025-04-03 11:33:12.749] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:162] [Co_PerformHttpRequest] Request /api/v1/events/stateless: Status 200
[2025-04-03 11:33:12.749] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:163] [Co_PerformHttpRequest] Request endpoint: /api/v1/events/stateless
Response: HTTP/1.1 200 OK
X-Imposter-Request: f4a27faf-7377-48b6-afaf-2b38933237cc
Server: imposter
content-length: 0


[2025-04-03 11:33:17.933] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:156] [SendMessageA] Message pushed: 'Security':'<Event xmlns='http://schemas.microsoft.com/win/2004/08/events/event'><System><Provider Name='Microsoft-Windows-Security-Auditing' Guid='{54849625-5478-4994-a5ba-3e3b0328c30d}'/><EventID>4625</EventID><Version>0</Version><Level>0</Level><Task>12544</Task><Opcode>0</Opcode><Keywords>0x8010000000000000</Keywords><TimeCreated SystemTime='2025-04-03T18:33:16.9137000Z'/><EventRecordID>32249</EventRecordID><Correlation ActivityID='{da50e5bc-a4fc-0000-a7e6-50dafca4db01}'/><Execution ProcessID='732' ThreadID='784'/><Channel>Security</Channel><Computer>DESKTOP-D42041D</Computer><Security/></System><EventData><Data Name='SubjectUserSid'>S-1-5-21-3444752446-47338941-2635532298-1001</Data><Data Name='SubjectUserName'>vagrant</Data><Data Name='SubjectDomainName'>DESKTOP-D42041D</Data><Data Name='SubjectLogonId'>0xa8d30</Data><Data Name='TargetUserSid'>S-1-0-0</Data><Data Name='TargetUserName'>vagrant</Data><Data Name='TargetDomainName'>DESKTOP-D42041D</Data><Data Name='Status'>0xc000006d</Data><Data Name='FailureReason'>%%2313</Data><Data Name='SubStatus'>0xc000006a</Data><Data Name='LogonType'>2</Data><Data Name='LogonProcessName'>seclogo</Data><Data Name='AuthenticationPackageName'>Negotiate</Data><Data Name='WorkstationName'>DESKTOP-D42041D</Data><Data Name='TransmittedServices'>-</Data><Data Name='LmPackageName'>-</Data><Data Name='KeyLength'>0</Data><Data Name='ProcessId'>0xb40</Data><Data Name='ProcessName'>C:\Windows\System32\svchost.exe</Data><Data Name='IpAddress'>::1</Data><Data Name='IpPort'>0</Data></EventData></Event>'
[2025-04-03 11:33:21.407] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:33:23.843] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:33:23.844] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\communicator\src\communicator.cpp:384] [ExecuteRequestLoop] Items count: 1
[2025-04-03 11:33:23.902] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:162] [Co_PerformHttpRequest] Request /api/v1/events/stateless: Status 200
[2025-04-03 11:33:23.902] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:163] [Co_PerformHttpRequest] Request endpoint: /api/v1/events/stateless
Response: HTTP/1.1 200 OK
X-Imposter-Request: ed657f2a-29fa-41f8-831a-b47953500ccf
Server: imposter
content-length: 0


[2025-04-03 11:33:24.086] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:156] [SendMessageA] Message pushed: 'System':'<Event xmlns='http://schemas.microsoft.com/win/2004/08/events/event'><System><Provider Name='EventLog'/><EventID Qualifiers='0'>1111</EventID><Version>0</Version><Level>4</Level><Task>1</Task><Opcode>0</Opcode><Keywords>0x80000000000000</Keywords><TimeCreated SystemTime='2025-04-03T18:33:24.0754346Z'/><EventRecordID>4899</EventRecordID><Correlation/><Execution ProcessID='824' ThreadID='0'/><Channel>System</Channel><Computer>DESKTOP-D42041D</Computer><Security/></System><EventData><Data>This is a test System event</Data></EventData></Event>'
[2025-04-03 11:33:29.549] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:156] [SendMessageA] Message pushed: 'Application':'<Event xmlns='http://schemas.microsoft.com/win/2004/08/events/event'><System><Provider Name='Application'/><EventID Qualifiers='0'>7777</EventID><Version>0</Version><Level>4</Level><Task>1</Task><Opcode>0</Opcode><Keywords>0x80000000000000</Keywords><TimeCreated SystemTime='2025-04-03T18:33:29.5388054Z'/><EventRecordID>5325</EventRecordID><Correlation/><Execution ProcessID='824' ThreadID='0'/><Channel>Application</Channel><Computer>DESKTOP-D42041D</Computer><Security/></System><EventData><Data>This is a test Application event</Data></EventData></Event>'
[2025-04-03 11:33:31.505] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:33:34.924] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\multitype_queue\src\multitype_queue.cpp:214] [getNextBytesAwaitable] Timeout reached after 10000ms
[2025-04-03 11:33:34.924] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\communicator\src\communicator.cpp:384] [ExecuteRequestLoop] Items count: 2
[2025-04-03 11:33:34.984] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:162] [Co_PerformHttpRequest] Request /api/v1/events/stateless: Status 200
[2025-04-03 11:33:34.984] [wazuh-agent] [trace] [TRACE] [C:\Users\wazuh-user\wazuh-agent\src\agent\http_client\src\http_client.cpp:163] [Co_PerformHttpRequest] Request endpoint: /api/v1/events/stateless
Response: HTTP/1.1 200 OK
X-Imposter-Request: 034abb0f-7dca-4051-8491-39e4e2584f2f
Server: imposter
content-length: 0


[2025-04-03 11:33:37.479] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\inventory\src\inventory.cpp:78] [Stop] Inventory module stopping...
[2025-04-03 11:33:37.480] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:204] [Wait] Logcollector coroutine timer was canceled.
[2025-04-03 11:33:37.480] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:204] [Wait] Logcollector coroutine timer was canceled.
[2025-04-03 11:33:37.480] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\winevt_reader\src\event_reader_win.cpp:86] [QueryEvents] Unsubscribing to channel 'System'.
[2025-04-03 11:33:37.481] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:204] [Wait] Logcollector coroutine timer was canceled.
[2025-04-03 11:33:37.481] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\winevt_reader\src\event_reader_win.cpp:86] [QueryEvents] Unsubscribing to channel 'Application'.
[2025-04-03 11:33:37.482] [wazuh-agent] [debug] [DEBUG] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:204] [Wait] Logcollector coroutine timer was canceled.
[2025-04-03 11:33:37.483] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\winevt_reader\src\event_reader_win.cpp:86] [QueryEvents] Unsubscribing to channel 'Security'.
[2025-04-03 11:33:37.491] [wazuh-agent] [info] [INFO] [C:\Users\wazuh-user\wazuh-agent\src\modules\logcollector\src\logcollector.cpp:101] [Stop] Logcollector module stopped.
```

--- a/docs/ref/modules/logcollector/README.md
+++ b/docs/ref/modules/logcollector/README.md
@@ -94,7 +94,7 @@ logcollector:
       query: Event[System/EventID = 37]
 ```
 
-This collector gets logs from the Windows Event Viewer. It needs a channel and a query to work. It will subscribe to the before mentioned channel with the specified query and it will return a json containing the events matching the criteria.
+This collector gets logs from the Windows Event Viewer. It needs a channel and a query to work (if this is empty '*' wildcard is assumed collecting all possible messages). It will subscribe to the before mentioned channel with the specified query and it will return a json containing the events matching the criteria.
 
 ```json
 {"agent":{"groups":[],"host":{"architecture":"x86_64","hostname":"HOSTNAME","ip":["LOCALIP","4444:4444:4444:4444:4444:44444:4444:4444","127.0.0.1","::1"],"os":{"name":"Microsoft Windows Server 2022","type":"Unknown","version":"10.0.20348.2762"}},"id":"4444-4444-4444-4444-ae5a7d59936c","name":"","type":"Endpoint","version":"x.y.z"}}
@@ -107,7 +107,7 @@ This collector gets logs from the Windows Event Viewer. It needs a channel and a
 |           | reload_interval | Time in milliseconds to recheck for subscriptions | 60000   |
 |     ✔️     | windows         | Vector of readers to subscribe                    |         |
 |     ✔️     | windows.channel | Channel name to be used for subscription          |         |
-|     ✔️     | windows.query   | Query to apply to the channel                     |         |
+|           | windows.query   | Query to apply to the channel                     |         |
 
 ### macOS (ULS) Collector
 

--- a/etc/config/wazuh_agent_linux.yml
+++ b/etc/config/wazuh_agent_linux.yml
@@ -25,7 +25,7 @@ logcollector:
   reload_interval: 1m
   read_interval: 500ms
   journald:
-    - field: "SYSLOG_IDENTIFIER"
-      value: "systemd"
-      exact_match: false
+    - field: PRIORITY
+      value: 0|1|2|3|4|5|6|7
+      exact_match: true
       ignore_if_missing: true

--- a/etc/config/wazuh_agent_macos.yml
+++ b/etc/config/wazuh_agent_macos.yml
@@ -23,6 +23,4 @@ logcollector:
   reload_interval: 1m
   read_interval: 500ms
   macos:
-    - query: process == "sshd" OR message CONTAINS "invalid"
-      level: info
-      type: trace,activity,log
+    - level: info

--- a/etc/config/wazuh_agent_windows.yml
+++ b/etc/config/wazuh_agent_windows.yml
@@ -25,10 +25,7 @@ logcollector:
   read_interval: 500ms
   windows:
     - channel: System
-      query: Event[System/EventID != 1000]
     - channel: Application
-      query: Event[System/EventID = 1000]
     - channel: Security
-      query: Event[System/EventID != 5145 and System/EventID != 5156 and System/EventID != 5447 and System/EventID != 4656 and System/EventID != 4658 and System/EventID != 4663 and System/EventID != 4660 and System/EventID != 4670 and System/EventID != 4690 and System/EventID != 4703 and System/EventID != 4907 and System/EventID != 5152 and System/EventID != 5157]
 
 

--- a/src/agent/configuration_parser/tests/configuration_parser_test.cpp
+++ b/src/agent/configuration_parser/tests/configuration_parser_test.cpp
@@ -319,14 +319,17 @@ TEST(ConfigurationParser, GetConfigOrDefaultArrayMap)
               api_token: api_token1
             - org_name: dummy2
               api_token: api_token2
+            - org_name: dummy3
     )";
     const auto parserStr = std::make_unique<configuration::ConfigurationParser>(strConfig);
-    const auto ret =
+    auto ret =
         parserStr->GetConfigOrDefault<std::vector<std::map<std::string, std::string>>>({}, "agent_array", "api_auth");
     ASSERT_EQ(ret[0].at("org_name"), "dummy1");
     ASSERT_EQ(ret[0].at("api_token"), "api_token1");
     ASSERT_EQ(ret[1].at("org_name"), "dummy2");
     ASSERT_EQ(ret[1].at("api_token"), "api_token2");
+    ASSERT_EQ(ret[2].at("org_name"), "dummy3");
+    ASSERT_EQ(ret[2]["api_token"], "");
 }
 
 TEST(ConfigurationParser, GetConfigOrDefaultMap)

--- a/src/modules/logcollector/src/winevt_reader/src/event_reader_win.cpp
+++ b/src/modules/logcollector/src/winevt_reader/src/event_reader_win.cpp
@@ -21,7 +21,7 @@ namespace logcollector::winevt
                                                        std::shared_ptr<IWinAPIWrapper> winAPI)
         : IReader(logcollector)
         , m_channel(channel)
-        , m_query(query)
+        , m_query(query.empty() ? "*" : query)
         , m_ChannelsRefreshInterval(channelRefreshInterval)
         , m_winAPI(winAPI ? winAPI : std::make_shared<DefaultWinAPIWrapper>())
     {

